scripts/release.js

```diff
@@ -0,0 +1,357 @@
+  if (hasMergeConflicts(resp)) {
+    let answer = await prompt(
+      `Merge conflicts detected. Resolve all conflicts and commit the changes before resuming the process.
+          ${chalk.bold("Press Y to continue or N to cancel the release.")}`
+    );
+    if (answer === false) return 0;
+  } else if (mergeFailed(resp)) {
+    console.error(chalk.red("Merge failed.\n"));
+    throw Error(resp);
+  }
+
+  execSync(`git checkout ${initialBranch}`);
+}
+
+/**
+ * @param {string} branch
+ * @returns
+ */
+async function gitPull(branch) {
+  try {
+    let resp = execSync(`git pull --rebase origin ${branch}`).toString();
+    if (hasMergeConflicts(resp)) {
+      let answer = await prompt(
+        `Merge conflicts detected. Resolve all conflicts and commit the changes before resuming the process.
+
+    ${chalk.bold("Press Y to continue or N to cancel the release.")}`
+      );
+      if (answer === false) return 0;
+    } else if (mergeFailed(resp)) {
+      console.error(chalk.red("Merge failed.\n"));
+      throw Error(resp);
+    }
+  } catch (e) {
+    console.error(chalk.red(`Error rebasing to origin/${branch}`));
+    throw e;
+  }
+}
+
+/**
+ * @param {string} currentVersion
+ * @param {string} givenVersion
+ * @param {string | undefined} [prereleaseId]
+ */
+function getNextVersion(currentVersion, givenVersion, prereleaseId = "pre") {
+  if (givenVersion == null) {
+    throw Error(
+      "Missing next version. Usage: node scripts/release.js start [nextVersion]"
+    );
+  }
+  // @ts-ignore
+  let nextVersion = semver.inc(currentVersion, givenVersion, prereleaseId);
+  if (nextVersion == null) {
+    throw Error(`Invalid version specifier: ${givenVersion}`);
+  }
+  return nextVersion;
+}
+
+function getCurrentBranch() {
+  let output = execSync("git rev-parse --abbrev-ref HEAD").toString().trim();
+  return output;
+}
+
+/**
+ * @param {string} output
+ * @returns {boolean}
+ */
+function hasMergeConflicts(output) {
+  let lines = output.trim().split("\n");
+  return lines.some(line => /^CONFLICT\s/.test(line));
+}
+
+/**
+ * @param {string} output
+ * @returns {boolean}
+ */
+function mergeFailed(output) {
+  let lines = output.trim().split("\n");
+  return lines.some(line => /^Automatic merge failed;\s/.test(line));
+}
+
+/**
+ * @param {string} branch
+ */
+function checkoutNewBranch(branch) {
+  let output = execSync(`git checkout -b ${branch}`).toString().trim();
+  if (/^fatal:/.test(output)) {
+    throw Error(`Branch ${chalk.bold(output)} already exists.`);
+  }
+}
+
+function getAllTags() {
+  return execSync("git tag --list").toString().trim().split("\n");
+}
+
+/**
+ * @param {string} branch
+ * @returns {"dev"}
+ */
+function ensureDevBranch(branch) {
+  if (branch !== "dev") {
+    throw Error("Releases should be created from the dev branch.");
+  }
+  return branch;
+}
+
+/**
+ * @param {string} branch
+ */
+function ensureReleaseBranch(branch) {
+  let version = getVersionFromReleaseBranch(branch);
```

Comment:
we'll want to make sure this is *the latest release branch* semver has some helpers that can do that I think.